src/org/thoughtcrime/securesms/ConversationFragment.java

```diff
@@ -424,13 +423,8 @@ private void handleCopyMessage(final Set<DcMsg> dcMsgsSet) {
         }
 
         if (result.length() > 0) {
-            try {
-                ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
-                clipboard.setText(result.toString());
-                Toast.makeText(getActivity(), getActivity().getResources().getString(R.string.copied_to_clipboard), Toast.LENGTH_LONG).show();
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            Util.writeTextToClipboard(getActivity(), result.toString());
+            Toast.makeText(getActivity(), getActivity().getResources().getString(R.string.copied_to_clipboard), Toast.LENGTH_LONG).show();
```

Comment:
This has the disadvantage that now we will show the "Copied to clipboard" toast unconditionally, even if copying failed. Everytime we call `writeTextToClipboard()`, we either show "Copied to clipboard" or "Done", so maybe we should let `writeTextToClipboard()` show the toast? (I think that "Copied to clipboard" would always fit)

Reply:
> even if copying failed

in theory now the copy doesn't fail, if I get it right it could fail because incompatible API, but now `Util.writeTextToClipboard()` uses the right API based on android version